ripple/src/main/webapp/businessgallery.js

```diff
@@ -20,26 +29,54 @@ function loadSearchResults() {
               const card = document.createElement('div');
               card.classList = 'row';
 
-            // Construct card content by appending HTML strings
-              var content = `
+               // Distance matrix calculations
+              // Get user's latitude and longitude from localStorage
+              var userLat = localStorage.getItem('enteredLat');
+              var userLong = localStorage.getItem('enteredLong');
+
+              // Get business latitude and longitude from GeoPoint object in firestore
+              var busLat = doc.data().coordinates.latitude;
+              var busLong = doc.data().coordinates.longitude;
+
+              // Reconstruct as maps LatLng object
+              var origin = new google.maps.LatLng(userLat, userLong);
+              var destination = new google.maps.LatLng(busLat, busLong);
+              
+              // Call distance matrix service
+              var service = new google.maps.DistanceMatrixService();
+              service.getDistanceMatrix(
+                {
+                  origins: [origin],
+                  destinations: [destination],
+                  travelMode: google.maps.TravelMode.WALKING,
+                }, callback);
+              
+              // In callback function, dynamically construct the HTML for the cards in each carousel
+              function callback(response, status) {
+                if (status !== "OK") {
+                    alert("Error with distance matrix");
+                    return;
+                }
+                // Construct card content by appending HTML strings
+                var content = `
                 <div class="col-6 col-md-4 col-lg-3 mb-4">
                     <div class="card h-100">
                     <img class="card-img-top img-fluid" id="card-dynamic-image" src="/serve?blob-key=${doc.data().thumbnailImage}">
                         <div class="card-body">
                             <p class="card-text">${doc.data().businessName[1]}</p>
-                            <p class="card-text"><small class="text-muted">${doc.data().walkingDistance}</small></p>
+                            <p class="card-text"><small class="text-muted">${response.rows[0].elements[0].duration.text + " walking"}</small></p>
```

Comment:
Just curious, is this how you access walking time from the distance matrix service?

Reply:
yeah